lib/srv/db/access_test.go

```diff
@@ -778,20 +778,29 @@ func TestAccessMongoDB(t *testing.T) {
 	}{
 		{
 			name: "client without compression",
-			opts: options.Client(),
+			opts: options.Client().
+				// Add extra time so the test won't time out when running in parallel.
+				SetServerSelectionTimeout(10 * time.Second),
```

Comment:
Increasing this timeout seems to help tests pass more often, but doesn't completely resolve the issue. I can still reproduce test failures locally under some CPU load (2 test processes running at the same time).

Reply:
I also managed to hit this once in 10 runs without running another test in parallel.
```
go test -race -run TestAccessMongoDB -count=10 ./lib/srv/db
--- FAIL: TestAccessMongoDB (0.01s)
    --- FAIL: TestAccessMongoDB/new_server/client_without_compression/no_access_to_databases (23.79s)
        access_test.go:819: 
            	Error Trace:	access_test.go:819
            	Error:      	"server selection error: server selection timeout, current topology: { Type: Single, Servers: [{ Addr: 127.0.0.1:42509, Type: Unknown, Average RTT: 0 }, ] }" does not contain "access to db denied"
            	Test:       	TestAccessMongoDB/new_server/client_without_compression/no_access_to_databases
FAIL
FAIL	github.com/gravitational/teleport/lib/srv/db	442.632s
FAIL
```

Reply:
I run this branch around 10 times in GCB now and this test hasn't failed even once. I think that there always will be a risk of this test being flaky, as we still rely on some kind of timeout anyways. The timeout is here already so I hope we don't add extra flakiness to it. In the worst case scenario we can always revert this PR :|